cmd/realm/triggers.go

```diff
@@ -177,3 +205,264 @@ func triggersDeleteF(command *cobra.Command, args []string) error {
+
+		y, err := command.Flags().GetBool("force")
+		if err != nil {
+			return err
+		}
+
+		if y {
+			fmt.Printf("The following triggers already exists and WILL be DELETED and RECREATED: %+q \n", list_existing)
+			fmt.Printf("\n")
+			if ok, err := utils.AskForConfirmation("Do you want to continue?"); !ok || err != nil {
+				fmt.Printf("aborting")
+				return nil
+			}
+
+			for _, v := range triggerToUpdate {
+				if err := updateTrigger(realm, v["name"].(string), v); err != nil {
+					fmt.Fprintf(os.Stderr, "Could not update trigger %s: %s\n", v["name"].(string), err)
+				} else {
+					fmt.Printf("trigger %s updated successfully\n", v["name"].(string))
+				}
+			}
+			fmt.Printf("\n")
+			fmt.Printf("\n")
+
+		} else {
+
+			// Start syncing.
+			fmt.Printf("The following triggers already exists and WILL NOT be updated: %+q \n", list_existing)
+			fmt.Printf("\n")
+
+		}
+	}
+
+	return nil
+}
+
+func installTrigger(realm string, trigger map[string]interface{}) error {
+	installTriggerCall, err := astarteAPIClient.InstallTrigger(realm, trigger)
+	if err != nil {
+		return err
+	}
+
+	// When we're here in the context of `interfaces sync`, the to-curl flag
+	// is always false (`interfaces sync` has no `--to-curl` flag)
+	// and thus the call will never exit unexpectedly
+	utils.MaybeCurlAndExit(installTriggerCall, astarteAPIClient)
+
+	installTriggerRes, err := installTriggerCall.Run(astarteAPIClient)
+	if err != nil {
+		return err
+	}
+
+	_, _ = installTriggerRes.Parse()
+	return nil
+}
+
+func updateTrigger(realm string, triggername string, newtrig map[string]interface{}) error {
+
+	deleteTriggercall, err := astarteAPIClient.DeleteTrigger(realm, triggername)
+	if err != nil {
+		return err
+	}
+	utils.MaybeCurlAndExit(deleteTriggercall, astarteAPIClient)
+
+	_, err = deleteTriggercall.Run(astarteAPIClient)
+	if err != nil {
+		return err
+	}
+
+	updateTriggerCall, err := astarteAPIClient.InstallTrigger(realm, newtrig)
+	if err != nil {
+		return err
+	}
+
+	// When we're here in the context of `interfaces sync`, the to-curl flag
+	// is always false (`interfaces sync` has no `--to-curl` flag)
+	// and thus the call will never exit unexpectedly
+	utils.MaybeCurlAndExit(updateTriggerCall, astarteAPIClient)
+
+	updateTriggerRes, err := updateTriggerCall.Run(astarteAPIClient)
+	if err != nil {
+		return err
+	}
+
+	_, _ = updateTriggerRes.Parse()
+	return nil
+}
+
+func listTriggers(realm string) ([]string, error) {
+	listTriggersCall, err := astarteAPIClient.ListTriggers(realm)
+	if err != nil {
+		return []string{}, err
+	}
+
+	utils.MaybeCurlAndExit(listTriggersCall, astarteAPIClient)
+
+	listTriggersRes, err := listTriggersCall.Run(astarteAPIClient)
+	if err != nil {
+		return []string{}, err
+	}
+	rawlistTriggers, err := listTriggersRes.Parse()
+	if err != nil {
+		return []string{}, err
+	}
+	return rawlistTriggers.([]string), nil
+}
+
+func getTriggerDefinition(realm, triggerName string) (map[string]interface{}, error) {
```

Comment:
Can you use this also inside `triggersShowF`? So that we don't duplicate code

Reply:
resolved in last commit